src/Raven.Server/Documents/Sharding/Handlers/Processors/Revisions/ShardedRevisionsHandlerProcessorForGetRevisions.cs

```diff
@@ -72,31 +74,33 @@ protected override async ValueTask GetRevisionsAsync(TransactionOperationContext
                 cmd = new GetRevisionsCommand(id, start, pageSize, metadataOnly);
             }
 
-            string etag = null; //TODO
-            if (NotModified(etag))
+            int shardNumber = RequestHandler.DatabaseContext.GetShardNumber(context, id);
+            var result = await RequestHandler.ExecuteSingleShardAsync(context, cmd, shardNumber, token);
+            var array = result.Results.Items.Select(x => ((BlittableJsonReaderObject)x).Clone(context)).ToArray();
```

Comment:
do you know why are we cloning here?

Reply:
@lastav5 can you answer this? :-)

Reply:
Not sure.. Probably a mistake. Doesn't seem like the cloning is needed now, if it ever was

Reply:
Done